pydantic/types.py

```diff
@@ -2429,3 +2437,92 @@ def __getattr__(self, item: str) -> Any:
                 return object.__getattribute__(self, item)
 
     __hash__ = object.__hash__
+
+
+_TAGGED_UNION_TAG_KEY = 'pydantic-tagged-union-tag'
```

Comment:
This should probably be moved to `pydantic._internal._core_utils`, like `NEEDS_APPLY_DISCRIMINATED_UNION_METADATA_KEY`, `HAS_INVALID_SCHEMAS_METADATA_KEY`, etc.

Reply:
Resolving, as I've moved this as requested